internal/internal_workflow.go

```diff
@@ -651,12 +670,14 @@ func (c *channelImpl) Close() {
 	}
 }
 
-// Takes a value and assigns that 'to' value.
-func (c *channelImpl) assignValue(from interface{}, to interface{}) {
+// Takes a value and assigns that 'to' value. logs a metric if it is unable to deserialize
+func (c *channelImpl) assignValue(from interface{}, to interface{}) (error) {
 	err := decodeAndAssignValue(c.dataConverter, from, to)
+	//add to metrics
 	if err != nil {
-		panic(err)
+		c.scope.Counter(metrics.CorruptedSignalsCounter).Inc(1)
```

Comment:
ditto

Reply:
The waiting happens in Receive method. I think assignValue should not be concerned with blocking and should only return an error

Reply:
resolved